src/tree/mod.rs

```diff
@@ -841,7 +854,24 @@ impl RatchetTree {
     }
 }
 
-pub type InvitationList = Vec<(NodeIndex, AddProposal)>;
+pub struct ApplyProposalsValues {
+    pub path_required: bool,
+    pub self_removed: bool,
+    pub invitation_list: Vec<(NodeIndex, AddProposal)>,
+}
+
+impl ApplyProposalsValues {
+    pub fn exclusion_list(&self) -> HashSet<&NodeIndex> {
```

Comment:
Can you add some documentation on what this is meant to do? The name is not very explanatory.